File: Ruby/rectangleRotation.rb
def rectangleRotation(a, b)
  ai = (a/Math.sqrt(2))/2
  b1 = (b/Math.sqrt(2))/2
  r1 = [ai.floor*2 + 1, b1.floor*2 + 1]
  r2 = []

  ai - ai.floor < 0.5 ? r2[0] = r1[0] - 1 : r2[0] = r1[0] + 1
  b1 - b1.floor < 0.5 ? r2[1] = r1[1] - 1 : r2[1] = r1[1] + 1

  p r1.inject(:*) + r2.inject(:*)
end

rectangleRotation(30, 2)

# Best solutions
# def rectangleRotation(a, b)
#     c = 1.414
#     a = (a / c).to_i
#     b = (b / c).to_i
#     c = ~(a ^ b) & 1
#
#     return 2 * a * b + a + b + c
# end

# or

File: Python/rectangleRotation.py
def rectangleRotation(a, b):
    r2=2**.5
    ax=int(a/r2)
    bx=int(b/r2)
    y=ax*bx+(ax+1)*(bx+1)
    y-= 0 if y%2 else 1
    return y

# Other solutions
# def rectangleRotation(a, b):
#     a = int(a//math.sqrt(2))
#     b = int(b//math.sqrt(2))
#
#     return odd(a ) * odd(b) + even(a ) * even(b )
#
# def odd(a):
#     a = int(a)
#     return 2 * (a // 2 + a % 2)
#
# def even(a):
#     a = int(a)
#     return 2 * (a // 2 ) + 1

# or
# def rectangleRotation(a, b):
#     [m, n] = [int(math.floor(x / math.sqrt(2))) for x in (a, b)]
#     return m * n + (m + 1) * (n + 1) - (m + n) % 2
